vncdotool/loggingproxy.py

```diff
@@ -55,11 +57,17 @@ def _handle_version(self):
                 self._handler = self._handle_clientInit, 1
         elif version in ('007', '008'):
             # XXX send security v3.7+
-            self._handler = self._handle_security, 1
+            self._handler = self._handle_security, 2
 
     def _handle_security(self):
-        sectype = self.buffer[0]
-        self.buffer = self.buffer[1:]
+        sectype = unpack('!H', self.buffer[:2])[0]
+        self.buffer = self.buffer[2:]
+        if sectype != 1:
+            self.transport.loseConnection()
+            print("sectype: ", sectype)
+            raise "Security type not implanted!"
```

Comment:
IIRC there was at least one case where just ignoring the security type worked, so this may break some previously working setups?

Reply:
Yeah, that looks like Monday coding for me!
That if statement should look like this:
```suggestion
            if sectype == 0:
                self._handler = self._handle_protocol, 1
```
Maybe a warning would be appropriate though?